hnswlib/bruteforce.h

```diff
@@ -106,6 +109,28 @@ namespace hnswlib {
             return topResults;
         };
 
+        template <typename Comp>
+        std::vector<std::pair<dist_t, labeltype>>
+        searchKnn(const void* query_data, size_t k, Comp comp) {
+            std::vector<std::pair<dist_t, labeltype>> result;
+            if (cur_element_count == 0) return result;
+
+            auto ret = searchKnn(query_data, k);
+
+            while (!ret.empty()) {
+                result.push_back(ret.top());
+                ret.pop();
+            }
+
+            if (result.size() > 1) {
```

Comment:
This additional loop can be avoided if filling the `result` from the back. This doesn't seem to be a big performance issue.

Reply:
Not quite understand. Can you show me how to do that?

Reply:
Something like:
```c++           
            if (cur_element_count == 0) {
                std::vector<std::pair<dist_t, labeltype>> result;
                return result;
            }
            
            auto ret = searchKnn(query_data, k);
            
            std::vector<std::pair<dist_t, labeltype>> result(ret.size);
            int i=ret.size-1;
            
            while (!ret.empty()) {
                result[i]=ret.top();
                ret.pop();
                i-=1;
            }
            return result;
```